core/aws_ddk_core/stages/appflow.py

```diff
@@ -0,0 +1,224 @@
+        state_machine_input : Optional[Dict[str, Any]]
+            Input of the state machine
+        """
+        super().__init__(scope, id)
+
+        self._state_machine_input: Optional[Dict[str, Any]] = state_machine_input
+        self._event_detail_type: str = f"{id}-event-type"
+
+        # If AppFlow flow name is not supplied, create one
+        self._flow: Optional[CfnFlow] = None
+        if not flow_name:
+            self._flow = CfnFlow(
+                self,
+                f"{id}-flow",
+                destination_flow_config_list=[destination_flow_config],
+                source_flow_config=source_flow_config,
+                tasks=tasks,
+                trigger_config=CfnFlow.TriggerConfigProperty(trigger_type="OnDemand"),
+            )
+            flow_name = self._flow.name
+
+        # Create lambda used to check flow execution status
+        status_lambda_role = Role(
+            self,
+            "flow-execution-status-lambda-role",
+            assumed_by=ServicePrincipal("lambda.amazonaws.com"),
+            description="lambda role to check appflow flow execution status",
+        )
+
+        status_lambda = LambdaFactory.function(
+            self,
+            id="flow-execution-status-lambda",
+            environment_id=environment_id,
+            code=Code.from_asset(str(Path(__file__).parent.joinpath("appflow_lambda/"))),
+            handler="index.lambda_handler",
+            role=status_lambda_role,
+        )
+        # Enable the function to get flow execution records
+        status_lambda.add_to_role_policy(
+            PolicyStatement(
+                effect=Effect.ALLOW,
+                actions=[
+                    "appflow:DescribeFlowExecutionRecords",
+                ],
+                resources=["*"],
+            )
+        )
+
+        # Create start flow step function task
+        flow_object = CustomState(
+            self,
+            "start-flow-execution",
+            state_json={
+                "Type": "Task",
+                "Resource": "arn:aws:states:::aws-sdk:appflow:startFlow",
+                "Parameters": {"FlowName": flow_name},
+                "Catch": [
+                    {
+                        "ErrorEquals": ["Appflow.ConflictException"],
+                        "Next": "wait-before-checking-flow-status-again",
+                    }
+                ],
+            },
+        )
+
+        # Create loop to check flow execution status
+        flow_execution_records = LambdaInvoke(
+            self,
+            "get-flow-execution-status",
+            lambda_function=status_lambda,
+            result_selector={"FlowExecutionStatus.$": "$.Payload"},
+        )
+
+        flow_object_execution_status = Choice(self, "check-flow-execution-status")
+        flow_object_execution_status_wait = Wait(
+            self,
+            "wait-before-checking-flow-status-again",
+            time=WaitTime.duration(Duration.seconds(15)),
```

Comment:
Should we make this timeout configurable?

Reply:
Leave it or take it for me. I don't think users would be that fussed about having control over this one and less params is better

Reply:
I think it highly depends on how long your flow takes to run. If it's running for an hour why would you check the status every 15 secs?

Reply:
added an argument called `flow_execution_status_check_period`, with a default value of 15 seconds